### DISCUSSION TITLE
CCAKEM IND-CCA2 vs CPAPEK in the KEM schemes.

### DISCUSSION THREAD

**Discussion by yerracs86** [open-quantum-safe/liboqs#1255] [Q&A] [answered]
Hi, I am a developer and trying to develop applications using algorithms using LibOQS. I have some basic question.

1. Why is there is only CCAKEM IND-CCA2 scheme but not CPAPKE scheme defined in the library for KEM methods.

Appreciate your help.

**Comment by dstebila** [comment, marked as answer]
Most submissions did not consider the CPA PKE version to be an official submission, only an intermediate building block for the CCA KEM version. Note that our SIKE implementation does include CPA version, SIDH. Generally it is risky to use the CPA versions except in very controlled scenarios where one can be certain there will be no reuse; it is safer to make only the CCA version available.
